pywbem_mock/_wbemconnection_mock.py

```diff
@@ -543,20 +545,94 @@ def compile_mof_string(self, mof_str, namespace=None, search_paths=None,
 
         mofcomp.compile_string(mof_str, namespace)
 
+    def compile_dmtf_classes(self, schema_mof_dir, schema_classnames,
```

Comment:
Instead of passing the path name of the schema directory in "schema_mof_dir" and using an implicitly defined file name of the pragma file, it would be more straight forward to pass the path name of the pragma file in a "schema_pragma_file" parameter.

Reply:
Actually, no it would not.  The schema file includes the schema version so that somewhere the version would have to be passed or the specific file for a schema version name defined.  That is why we did it this way.

**DISCUSSION**

Reply:
The schema file would be passed all the way through all layers. And yes, at the top its file name needs to be known. A CIM schema that is not the DMTF CIM schema might not even have the version in the file name. Or it may name the file differently from the current glob pattern.

Reply:
Either way you are making assumptions.  1. Your assume that the schema file in in the schema_mof_dir or 2. you assume that the schema_mof_pragma_file directory can be used as the search path.

I would argue that the that the schema_mof_dir is more general that the schema_mof_pragma_file but in either case you end up with a name that may include a schema version.

DISCUSSION

Reply:
On assumptions:
> 1. Your assume that the schema file in in the schema_mof_dir or 2. you assume that the schema_mof_pragma_file directory can be used as the search path.

On 1.: The path name of the schema file contains its directory, so a separate schema_mof_dir parameter is not needed to find it.
On 2.: The pragma include statements in the schema file reference the other files, and if needed, with a relative path name, so the schema file is self-contained w.r.t. the location of other files.

Where the assumption really is, is by assuming the class mof files have a file name that can be determined from the CIM class name. That is true for the DMTF schema since schema version 2.7 or so (but not before), and it may or may not be true for other CIM schemas. That assumption does not take advantage of the fact that by just knowing the schema file everything is determined, and it requires knowing a directory under which the class mof files are located so they can be found by constructing their file name from the desired class name.

I think it boils down to the question what our requirements on the file structure of a CIM schema should be, and how general we can/want to get. At the one extreme, we would only support the exact structure of today's DMTF CIM schema. At the other extreme, we would support any MOF-legal way of including files into one top level schema file.

Reply:
We discuss this in our call and between the two options of pointing to a schema directory and pointing to the pragma file, Andy has a slight preference to point to the pragma file, which is what the latest commit does.